sdk/core/ci.yml

```diff
@@ -11,7 +11,9 @@ trigger:
       - sdk/core/
       - sdk/parents/azure-sdk-parent/
       - sdk/parents/azure-client-sdk-parent/
-      - eng/
```

Comment:
We might want to just exclude eng/versioning/ instead of splitting this up like this. If we keep this as include filters we should also add eng/common to the include list.

Reply:
Currently there is more to exclude than include under `/eng` which is why I went this direction. Can add `/common` to the list of includes in the next commit.

Reply:
Hrm, I'm with Wes on this actually because one of the reasons we have core trigger on eng changes is to pick up issues when there are changes there. By using includes rather than excludes we may potentially miss things that are added over time.

If the cause for thrashing is changes in the eng/versioning directory then we should just exclude that.

Reply:
Updated to use @weshaggard's pattern.

Reply:
I also added `eng/jacoco-test-coverage` as this generates a lot of CI churn as it will update when `eng/versioning` updates without directly affecting Core.